src/scheduling/cost_models/simulated_quincy_cost_model.cc

```diff
@@ -0,0 +1,358 @@
+	// (costs are instead encoded in arc from task to aggregator)
+	return 0LL;
+}
+
+Cost_t SimulatedQuincyCostModel::EquivClassToEquivClass(EquivClass_t tec1,
+                                                        EquivClass_t tec2) {
+	// this shouldn't be called; the only arcs are from cluster aggregator to rack
+	// aggregator, but Firmament considers cluster aggregator to be a special case
+	CHECK(false);
+  return 0LL;
+}
+
+// In Quincy, a task is in its own equivalence class
+vector<EquivClass_t>* SimulatedQuincyCostModel::GetTaskEquivClasses(
+    TaskID_t task_id) {
+  vector<EquivClass_t>* preferred_res = new vector<EquivClass_t>();
+  auto &preferred_racks = preferred_rack_map_[task_id];
+  for (auto elt : preferred_racks) {
+    EquivClass_t rack = elt.first;
+    VLOG(1) << "Task " << task_id << " has arc to rack aggregator " << rack;
+    preferred_res->push_back(rack);
+  }
+  return preferred_res;
+}
+
+vector<EquivClass_t>* SimulatedQuincyCostModel::GetResourceEquivClasses(
+    ResourceID_t res_id) {
+	vector<EquivClass_t>* equiv_classes = new vector<EquivClass_t>();
+	EquivClass_t rack_aggregator = machine_to_rack_map_[res_id];
+	equiv_classes->push_back(rack_aggregator);
+	VLOG(1) << "Machine " << res_id << " belongs to rack " << rack_aggregator;
+	return equiv_classes;
+}
+
+vector<ResourceID_t>* SimulatedQuincyCostModel::GetOutgoingEquivClassPrefArcs(
+    EquivClass_t tec) {
+  vector<ResourceID_t>* res = new vector<ResourceID_t>();
+  return res;
+}
+
+vector<TaskID_t>* SimulatedQuincyCostModel::GetIncomingEquivClassPrefArcs(
+    EquivClass_t tec) {
+  vector<TaskID_t>* res = new vector<TaskID_t>();
+  return res;
+}
+
+vector<ResourceID_t>* SimulatedQuincyCostModel::GetTaskPreferenceArcs(TaskID_t task_id) {
+  vector<ResourceID_t>* preferred_res = new vector<ResourceID_t>();
+  auto &preferred_machines = *(preferred_machine_map_[task_id]);
+  for (auto elt : preferred_machines) {
+  	ResourceID_t machine = elt.first;
+  	preferred_res->push_back(machine);
+  	VLOG(1) << "Task " << task_id << " has preference arc to machine " << machine;
+  }
+  return preferred_res;
+}
+
+pair<vector<EquivClass_t>*, vector<EquivClass_t>*>
+    SimulatedQuincyCostModel::GetEquivClassToEquivClassesArcs(EquivClass_t tec) {
+	vector<EquivClass_t>* incoming_arcs = new vector<EquivClass_t>();
+	vector<EquivClass_t>* outgoing_arcs = new vector<EquivClass_t>();
+	return pair<vector<EquivClass_t>*, vector<EquivClass_t>*>
+		 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 	 (incoming_arcs, outgoing_arcs);
+}
+
+void SimulatedQuincyCostModel::AddMachine(
+    ResourceTopologyNodeDescriptor* rtnd_ptr) {
+	// we use ResourceID_t to identify machines
+	ResourceID_t res_id = ResourceIDFromString(rtnd_ptr->resource_desc().uuid());
+	// 'replicate' blocks
+	filesystem_->addMachine(res_id);
+	// bin it into a rack
+	EquivClass_t current_rack = rack_to_machine_map_.size() - 1;
+	if (rack_to_machine_map_[current_rack].size() >= machines_per_rack_) {
```

Comment:
Won't fix for this PR. Have added a comment explaining what's going on here and pointing out that we should change it.